src/read.c

```diff
@@ -4107,6 +4106,10 @@ avifResult avifDecoderReset(avifDecoder * decoder)
                 avifDiagnosticsPrintf(data->diag, "Box[tmap] gain map item ID %d not found", gainMapItemID);
                 return AVIF_RESULT_INVALID_TONE_MAPPED_IMAGE;
             }
+            if (avifDecoderItemShouldBeSkipped(gainMapItem)) {
+                avifDiagnosticsPrintf(data->diag, "Box[tmap] gain map item %d is not a supported image type", gainMapItemID);
+                return AVIF_RESULT_INVALID_TONE_MAPPED_IMAGE;
+            }
```

Comment:
Nit: An error is returned but the function is called `avifDecoderItemShouldBeSkipped()`. Maybe we should consider changing the behavior or renaming the function in the future but this is fine for now.

Reply:
Good point, will fix in another PR.